orchagent/portsorch.cpp

```diff
@@ -3704,8 +3704,15 @@ void PortsOrch::doLagMemberTask(Consumer &consumer)
 
             if (lag.m_members.find(port_alias) == lag.m_members.end())
             {
-                /* Assert the port doesn't belong to any LAG already */
-                assert(!port.m_lag_id && !port.m_lag_member_id);
+                /* Assert the port is not a LAG */
+                assert(port.m_lag_id == SAI_NULL_OBJECT_ID);
+
+                if (port.m_lag_member_id != SAI_NULL_OBJECT_ID)
+                {
+                    SWSS_LOG_NOTICE("Port %s is already a LAG member", port.m_alias.c_str());
```

Comment:
IMO, please move it to INFO as we dont want this possible retry code to continuously spew logs. Otherwise the change lgtm. @judyjoseph to sign-off.

Reply:
This will not continuously print in logs, when the situation described above happens, you'll get only 1 notice message. I did not want to make it lower level because if user configured lag member which is already a part of different lag he will not see any notification in the log.

Reply:
But you are not erasing the object, right? So there will be a retry in the next doTask? Can you please confirm?

Reply:
I think Prince is referring to a case where Ethernet0 is already part of a different Port Channel & user issues this command only  --> config portchannel member add PortChannel0001 Ethernet0;   (no delete command issued )

Reply:
Does this mean orchagent should not print anything on such invalid configuration?

Reply:
@judyjoseph , in the above case, IMO, the CLI itself should return error and should not accept the command. My concern is about a race condition, which we expect to get resolved in next few retries. @stepanblyschak , afaik, this is not an invalid configuration. If you say its invalid config, then error should be returned from upper layers and not orchagent.

Reply:
@prsunny "error should be returned from upper layers" - this is correct for CLI only for >=202012. There is a check in CLI whether port is already a lag member. In 201911 there is no such check, also a "config reload" check does not exist in any release meaning we either have to:
1. Don't print anything to log by default (INFO or DEBUG) making it harder for end user to figure out there is something wrong with configuration.
2. Have a one time NOTICE message in case the scenario described above happens which is rare and in case of invalid configuration user will see a lot of failed retries in the log rather then nothing.

Reply:
so this fix is only intended for 202012 and master, right. I dont see it marked for 201911 (although i'm *not* suggesting for 201911)

Reply:
> in case of invalid configuration user will see a lot of failed retries in the log rather then nothing

In this case, infinite retry and logging are bad, the best effort orchagent could do:
1. error should be returned from upper layers
2. discard the invalid configuration and do not retry infintely

Reply:
@qiluo-msft Then how to deal with reorder? or it is a ConsumerStateTable bug? although ConsumerStateTable always worked without preserving order as far as I know.

Reply:
@prsunny I suggest mark it for 201911 as well as I see a possibility for this to happen there as well.

Reply:
I am only comment on "infinite retry". For reorder case, I think retry will pass finally.

Reply:
I dont think this is required for 201911 as it is relatively stable. Secondly, please fix the log level as discussed above. As mentioned, if it is a reorder case, it will pass.. if not it will infinitely spew log.

Reply:
@prsunny changed. Regarding 201911 I just want to make it clear that it is possible althoug I didn't tested it (a reporduction may require to put some artificial load on CPU while doing member add/remove in the loop). In 201911 it is even worth then in 202012. In 201911 the call is done in sairedis async mode which means it will crash orchagent if the call is not successful. In 202012 it will call SAI but orchagent will retry later.